beta-prompts/beta_prompts/generation_system.py

```
@@ -0,0 +1,955 @@
+                    else:
+                        variables_used[var_name] = random.choice(var_options)
+                else:
+                    variables_used[var_name] = random.choice(var_options)
+
+        return variables_used
+
+    def _generate_from_template(self, template: str, variables: dict[str, str]) -> str:
+        """Generate prompt from template with variable substitution"""
+
+        # Use safe template substitution that ignores missing variables
+        import string
```

Ruff (0.12.2)

`import` should be at the top-level of a file

(PLC0415)